dspace-api/src/main/java/org/dspace/storage/rdbms/DatabaseUtils.java

```diff
@@ -538,7 +540,7 @@ private synchronized static Flyway setupFlyway(DataSource datasource)
      * @throws SQLException if database error
      *      If database cannot be upgraded.
      */
-    public static synchronized void updateDatabase()
```

Comment:
Note: There are *two* `updateDatabase()` methods (with different params) in this class, and it seems you only removed the `synchronized` from one of them. Is that on purpose?

Reply:
I think that currently we synchronise our threads a bit too much. I think this results in all REST API calls to go sequentially through this lock. Additionally, we might think about not checking Flyway for each and every REST API call. But I'll take that discussion to another ticket. I've reverted the change here.

Reply:
I've created a ticket for this: https://jira.duraspace.org/browse/DS-3781